pywbem/cim_obj.py

```diff
@@ -2853,6 +2853,124 @@ def tomof(self, indent=0, maxline=MAX_MOF_LINE):
+        optionally set the path component of the CIMInstance.  If any of the
+        properties in the class have default values, those values are passed
+        to the instance unless overridden by the property_values dictionary.
+        No CIMProperty qualifiers are included in the created instance and the
+        `class_origin` attribute is transfered from the class only if the
+        `include_class_origin` parameter is True
+
+        Parameters:
+          klass (:class:`pywbem:CIMClass`)
+            CIMClass from which the instance will be constructed.  This
+            class must include qualifiers and should include properties
+            from any superclasses to be sure it includes all properties
+            that are to be built into the instance. Properties may be
+            excluded from the instance by not including them in the `klass`
+            parameter.
+
+          namespace (:term:`string`):
+            Namespace in the WBEMConnection used to retrieve the class or
+            `None` if the default_namespace is to be used.
+
+          property_values (:class:`py:dict` or `NocaseDict`_):
+            Dictionary containing name/value pairs where the names are the
+            names of properties in the class and the properties are the
+            property values to be set into the instance. The values must match
+            the type defined for the property in the class. If a property is in
+            the property_values dictionary but not in the class a ValueError
+            exception is raised. Property name case independence is only
+            honored if it is a NoCaseDict (`NocaseDict`_).
+
+          include_null_properties (:class:`py:bool`):
+            Determines if properties not in the property_values dictionary are
+            included in the instance. Default is `True`.
+
+            If `True` they are included in the instance returned with any
+            value from the class.
+
+            If `False` they are not included in the instance returned.
+
+         include_class_origin  (:class:`py:bool`):
+            Determines if ClassOrigin information is included in the returned
+            instance. Default is `False`.
+
+            If None or False, class origin information is not included.
+
+            If True, class origin information is included.
+
+          include_path (:class:`py:bool`:):
+            If `True` the CIMInstanceName path is build and inserted into
+            the new instance.  If `strict` all key properties must be in the
+            instance. Default is `True`
+
+          strict (:class:`py:bool`:):
+            If `True` and `include_path` is set, all key properties must be in
+            the instance so that a correct path can be built. Default is
+            `False`.
+
+            If not `True` The path component is created even if not all
+            key properties are in the created instance.
```

Comment:
Describe how the missing key properties are set in the CIMInstanceName object. Set to NULL? Missing?

Reply:
DONE: They are missing.

Reply:
Note that this is actually in from_instance code